Only report change when home directory is different on FreeBSD

samdoran opened the pull request:
##### SUMMARY

Fix regression introduced in c1e4ef39cb8d33dcb59a1b0a4984b16623d2cc01 that resulted in the module always reporting a change if the home directory was specified.

Fixes #42484 

Add tests to cover this scenario.

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
`user.py`

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
2.6
2.7
```

samdoran committed:
Only report change when home directory is different

Add tests with home: parameter

mattclay commented:
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/74415/24/tests

samdoran committed:
Fix tests

Have to skip macOS for now since there is a bug when specifying the home directory path for an existing user that results in a module failure. That needs to be fixed in a separate PR.

samdoran commented:
Test failure revealed a bug in the Darwin class, unrelated to this PR. I'll make a note to fix that in another PR.
